src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclaratorHandler.java

```diff
@@ -112,22 +121,60 @@ private FieldDeclaration handleFieldDeclarator(CPPASTDeclarator ctx) {
       initializer = lang.getInitializerHandler().handle(init);
     }
 
-    // type will be filled out later
-    FieldDeclaration declaration =
-        NodeBuilder.newFieldDeclaration(
-            ctx.getName().toString(),
-            UnknownType.getUnknownType(),
-            emptyList(),
-            ctx.getRawSignature(),
-            this.lang.getLocationFromRawNode(ctx),
-            initializer,
-            true);
+    String name = ctx.getName().toString();
+
+    FieldDeclaration declaration;
+
+    if (name.contains(lang.getNamespaceDelimiter())) {
+      String[] rr = name.split(lang.getNamespaceDelimiter());
+
+      String recordName =
+          String.join(lang.getNamespaceDelimiter(), Arrays.asList(rr).subList(0, rr.length - 1));
+      String fieldName = rr[rr.length - 1];
+
+      declaration =
+          NodeBuilder.newFieldDeclaration(
+              fieldName,
+              UnknownType.getUnknownType(),
+              emptyList(),
+              ctx.getRawSignature(),
+              this.lang.getLocationFromRawNode(ctx),
+              initializer,
+              true);
+
+      var recordDeclaration =
+          this.lang
+              .getScopeManager()
+              .getRecordForName(this.lang.getScopeManager().getCurrentScope(), recordName);
+
+      // prepared for PR #223 - to set the definition here
```

Comment:
@konradweiss This would be the place to add the definition magic for external static fields, if it is still intended.

Reply:
Currently the associated tests don't fail so I wonder why. I will take a look into it.

Reply:
@oxisto I finally got around to integrate the function pointer resolution of fptrs that do not reference a class over the scope manager by integrating a part of the function pointer resolution. If this is the change you accept we can finally merge.